readthedocs/projects/models.py

```diff
@@ -610,18 +611,17 @@ def get_production_media_path(self, type_, version_slug, include_file=True):
     def get_production_media_url(self, type_, version_slug, full_path=True):
         """Get the URL for downloading a specific media file."""
         try:
-            path = reverse(
-                'project_download_media',
-                kwargs={
-                    'project_slug': self.slug,
-                    'type_': type_,
-                    'version_slug': version_slug,
-                },
-            )
+            # NOTE: we can't use ``reverse('project_download_media')`` here
+            # because this URL only exists in El Proxito and this method is
+            # accessed from Web instance
+            path = f'/{DOC_PATH_PREFIX}downloads/{type_}/{self.slug}/{version_slug}/'
         except NoReverseMatch:
             return ''
         if full_path:
-            path = '//{}{}'.format(settings.PRODUCTION_DOMAIN, path)
+            # Use project domain for full path --same domain as docs
+            # (project-slug.{PUBLIC_DOMAIN} or docs.project.com)
+            domain = self.subdomain()
+            path = f'//{domain}{path}'
         return path
```

Comment:
Will `path` be wrong if we haven't specified `full_path`? Where are we returning `path` and not using `full_path`?

Reply:
Yes. There is no reason to not return the full path. This method is no used with `full_path=False` anyways. I refactored this method.